test/lib/getWorkflow.test.js

```diff
@@ -21,72 +21,179 @@ describe('getWorkflow', () => {
+            jobs: {
+                foo: { requires: ['~commit'] },
+                A: { requires: ['foo'] },
+                B: { requires: ['foo', 'stage@other:teardown'] },
+                C: { requires: ['stage@deploy:setup'], stage: { name: 'deploy', startFrom: true } },
+                D: { requires: ['C'], stage: { name: 'deploy' } },
+                E: { requires: ['stage@test:setup'], stage: { name: 'test', startFrom: true } },
+                main: { requires: ['stage@other:setup'], stage: { name: 'other', startFrom: true } },
+                publish: { requires: ['main'], stage: { name: 'other' } },
+                'stage@other:setup': { requires: [], stage: { name: 'other' } },
+                'stage@other:teardown': { stage: { name: 'other' } },
+                'stage@deploy:setup': { requires: ['A'], stage: { name: 'deploy' } },
+                'stage@deploy:teardown': { stage: { name: 'deploy' } },
+                'stage@test:setup': { requires: ['foo'], stage: { name: 'test' } },
+                'stage@test:teardown': { stage: { name: 'test' } }
+            },
+            stages: {
+                other: {
+                    setup: {},
+                    description: 'For canary deployment',
+                    jobs: ['main', 'publish'],
+                    startFrom: 'main'
+                },
+                deploy: {
+                    requires: ['A'],
+                    jobs: ['C', 'D']
+                },
+                test: {
+                    requires: 'foo',
+                    jobs: ['E']
                 }
+            }
+        };
+        const result = await getWorkflow({
+            pipelineConfig,
+            triggerFactory: triggerFactoryMock,
+            pipelineOnly: false
+        });
+
+        assert.deepEqual(result.pipelineWorkflow, {
+            nodes: [
+                { name: '~pr' },
+                { name: '~commit' },
+                { name: 'foo' },
+                { name: 'A' },
+                { name: 'B' },
+                { name: '~stage@other' },
+                { name: '~stage@deploy' },
+                { name: '~stage@test' }
+            ],
+            edges: [
+                { src: '~commit', dest: 'foo' },
+                { src: 'foo', dest: 'A' },
+                { src: 'foo', dest: 'B', join: true },
+                { src: 'stage@other', dest: 'B', join: true },
+                { src: 'A', dest: 'stage@deploy' },
+                { src: 'foo', dest: 'stage@test' }
+            ]
+        });
+        assert.deepEqual(result.stageWorkflows, {
+            other: {
+                nodes: [
+                    { name: 'stage@other:teardown' },
+                    { name: 'main' },
+                    { name: 'stage@other:setup' },
+                    { name: 'publish' }
+                ],
+                edges: [
+                    { src: 'stage@other:setup', dest: 'main' },
+                    { src: 'main', dest: 'publish' }
```

Comment:
Shouldn't there be an edge from leaf nodes ([`publish`]) to `teardown`?
```suggestion
                    { src: 'stage@other:setup', dest: 'main' },
                    { src: 'main', dest: 'publish' },
                     { src: 'publish', dest: 'stage@other:teardown' }
```

Reply:
Also the actual result (below) is not matching the expectation
```
[
  {
    "src": "main",
    "dest": "publish"
  }
]
```